### PR TITLE
Add format checkbox when importing mount points

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/S2PrGCQz/388-ostumbleweed-p5-1094924-partition-imported-during-installation-would-all-be-formatted-by-default

### CODE REVIEW

File: src/lib/y2partitioner/actions/controllers/fstabs.rb [hunk 1]
```diff
@@ -104,6 +109,9 @@ def import_mount_points
 
       private
 
+        SYSTEM_VOLUMES_MOUNT_POINTS = ["/", "/usr", "/var", "/opt", "/boot", "/boot/zipl"].freeze
```
Comment (jreidinger) [inline src/lib/y2partitioner/actions/controllers/fstabs.rb]: on which is this based? Please add comment explaining what it is.

File: src/lib/y2partitioner/actions/controllers/fstabs.rb [hunk 2]
```diff
@@ -173,12 +185,30 @@ def can_be_imported?(entry)
...
+          system_volume_mount_point?(mount_point) || !device.formatted?
+        end
+
+        def system_volume_mount_point?(mount_point)
+          system_volume_mount_points.include?(mount_point)
+        end
+
+        def system_volumes_mount_points
+          mount_points = SYSTEM_VOLUMES_MOUNT_POINTS.dup
+          mount_points.delete("/boot/zipl") unless arch.s390?
```
Comment (jreidinger) [inline src/lib/y2partitioner/actions/controllers/fstabs.rb]: what about different approach and having arch specific array. which is concatenated for given architecture?

File: src/lib/y2partitioner/actions/controllers/fstabs.rb [hunk 3]
```diff
@@ -35,6 +36,10 @@ class Fstabs
         # @return [Y2Storage::Fstab] fstab file selected to import mount points
         attr_accessor :selected_fstab
 
+        # @return [Boolean] whether the system partition must be formatted
```
Comment (jreidinger) [inline src/lib/y2partitioner/actions/controllers/fstabs.rb]: what is definition of system partition?

File: src/lib/y2partitioner/actions/controllers/fstabs.rb [hunk 4]
```diff
@@ -127,6 +140,13 @@ def disk_analyzer
           DeviceGraphs.instance.disk_analyzer
         end
 
+        # Current system architecture
+        #
+        # @return [Storage::Arch]
+        def arch
+          @arch ||= Y2Storage::StorageManager.instance.arch
+        end
```
Comment (ancorgs) [inline src/lib/y2partitioner/actions/controllers/fstabs.rb]: Do we need something that `Yast::Arch` cannot provide? If not, I would use `Yast::Arch` (just like the rest of YaST and most places of y2-storage-ng). The StorageManager arch is only used in y2-storage-ng when `Arch#efiboot?` if needed. In all other places it uses `Yast::Arch`.

File: src/lib/y2storage/blk_device.rb [hunk 5]
```diff
@@ -79,6 +79,12 @@ class BlkDevice < Device
     storage_forward :size, as: "DiskSize"
     storage_forward :size=
 
+    # @!method active?
+    #   Checks whether the device is active (LV, Encryption, etc)
```
Comment (ancorgs) [inline src/lib/y2storage/blk_device.rb]: It could be better documented what "active" means.

File: src/lib/y2storage/crypttab.rb [hunk 6]
```diff
@@ -31,14 +31,19 @@ class Crypttab
     CRYPTTAB_PATH = "/etc/crypttab"
     private_constant :CRYPTTAB_PATH
 
+    # @return [Filesystems::Base]
```
Comment (ancorgs) [inline src/lib/y2storage/crypttab.rb]: NP: documentation. To know what `#filesystem` is really about I need to check the documentation of the constructor, which is not obvious from here.

I would document here what the attribute is about, not in the constructor.

File: test/y2storage/devicegraph_test.rb [hunk 7]
```diff
@@ -391,19 +391,21 @@ def with_sda2_deleted(initial_graph)
 
       it "does not perform a system lookup" do
         expect(Y2Storage::BlkDevice).to_not receive(:find_by_any_name)
+        devicegraph.find_by_any_name("/dev/sda1")
```
Comment (ancorgs) [inline test/y2storage/devicegraph_test.rb]: :+1: Good catch about these broken tests!... which is kind of embarrassing, by the way

File: src/lib/y2partitioner/actions/controllers/fstabs.rb [hunk 8]
```diff
@@ -267,8 +330,56 @@ def setup_blk_filesystem(filesystem)
...
+          Y2Storage::Encryption.use_crypttab_names(fixed_devicegraph, crypttab)
+          fixed_devicegraph
+        end
+
+        # Selects the crypttab contained in the same filesystem than the selected fstab
+        #
+        # @return [Y2Storage::Crypttab, nil] nil if the filesystem does not contain
+        #   a crypttab file.
+        def crypttab
+          return @crypttab if @crypttab_found
```
Comment (ancorgs) [inline src/lib/y2partitioner/actions/controllers/fstabs.rb]: Hmmm. Does this mean the crypttab is not refreshed even if `#selected_fstab` changes? Looks like a bug to me.

Comment (ancorgs) [reply src/lib/y2partitioner/actions/controllers/fstabs.rb]: What about creating a unit test case for that corner case? Just in case some other developer feel tempted to cache that in the future. Only if it's easy enough, of course.
